### ISSUE TITLE
`map<string, MyEnum>` generates incorrect output when `stringEnums=true`

### ISSUE THREAD

**Issue by raymondpebble** [stephenh/ts-proto#543]
The generated code looks like this:

```
  fromJSON(object: any): MyMessage {
    const message = {...base} as MyMessage;
    message.mapOfEnums = Object.entries(object.mapOfEnums ?? {}).reduce<{[key: string]: MyEnum}>(
      (acc, [key, value]) => {
        acc[key] = value as number;
        return acc;
      },
      {},
    )
```

Assigning `value as number` to `acc[key]` is an error when using string enums.

**Comment by stephenh** [comment]
@raymondpebble thanks for filing; I think that this was _just_ fixed by #529 ? I'm pretty sure enough that I'm going to close this as a duplicate, but let me know if not. Thanks!

**Comment by raymondpebble** [comment]
Oh awesome, thank you! Sorry I didn't see that :)
